models/classes/class.TestsService.php

```diff
@@ -344,31 +338,49 @@ public function setTestModel(core_kernel_classes_Resource $test, core_kernel_cla
 					$former->deleteContent($test);
 				}	
 			}
-			$test->editPropertyValues(new core_kernel_classes_Property(self::PROPERTY_TEST_TESTMODEL), $testModel);
+			$test->editPropertyValues($this->getProperty(self::PROPERTY_TEST_TESTMODEL), $testModel);
 			$newImpl = $this->getTestModelImplementation($testModel);
 			if (!empty($newImpl)) {
 				$newImpl->prepareContent($test, $items);
 			}
 		}
     }
 
+    /**
+     * Returns a compiler instance for a given test
+     * @param core_kernel_classes_Resource $test
+     * @param ServiceFileStorage $storage
+     * @return tao_models_classes_Compiler
+     */
+    public function getCompiler(core_kernel_classes_Resource $test, ServiceFileStorage $storage) {
```

Comment:
ServiceFileStorage is not a used namespace

Reply:
the use statement is in line 28: oat\tao\model\service\ServiceFileStorage;

Reply:
my bad I didn't have the class at all, this is why it showed error :)